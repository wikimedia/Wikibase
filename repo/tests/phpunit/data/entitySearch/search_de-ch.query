{
	"search": "Wien",
	"language": "de-ch",
	"type": "item",
	"strictlanguage": false
}
