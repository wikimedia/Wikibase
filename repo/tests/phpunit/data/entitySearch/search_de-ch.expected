{
    "description": "wikibase_prefix search for 'Wien'",
    "params": {
        "timeout": "20s"
    },
    "query": {
        "query": {
            "bool": {
                "should": [
                    {
                        "bool": {
                            "filter": [
                                {
                                    "match": {
                                        "labels_all.prefix": "Wien"
                                    }
                                }
                            ],
                            "should": [
                                {
                                    "dis_max": {
                                        "tie_breaker": 0,
                                        "queries": [
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de-ch.near_match": "Wien"
                                                        }
                                                    },
                                                    "boost": 1.7
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de-ch.near_match_folded": "Wien"
                                                        }
                                                    },
                                                    "boost": 1.1
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de-ch.prefix": "Wien"
                                                        }
                                                    },
                                                    "boost": 0.9
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels_all.near_match_folded": "Wien"
                                                        }
                                                    },
                                                    "boost": 0.7
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de.near_match": "Wien"
                                                        }
                                                    },
                                                    "boost": 1.53
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de.near_match_folded": "Wien"
                                                        }
                                                    },
                                                    "boost": 0.99
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de.prefix": "Wien"
                                                        }
                                                    },
                                                    "boost": 0.81
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.near_match": "Wien"
                                                        }
                                                    },
                                                    "boost": 1.377
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.near_match_folded": "Wien"
                                                        }
                                                    },
                                                    "boost": 0.891
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.prefix": "Wien"
                                                        }
                                                    },
                                                    "boost": 0.729
                                                }
                                            }
                                        ]
                                    }
                                }
                            ]
                        }
                    },
                    {
                        "term": {
                            "title.keyword": "Wien"
                        }
                    }
                ],
                "minimum_should_match": 1,
                "filter": [
                    {
                        "term": {
                            "content_model": "wikibase-item"
                        }
                    }
                ]
            }
        },
        "_source": [
            "namespace",
            "title",
            "labels.de-ch",
            "descriptions.de-ch",
            "labels.de",
            "descriptions.de",
            "labels.en",
            "descriptions.en"
        ],
        "stored_fields": [],
        "highlight": {
            "pre_tags": [
                ""
            ],
            "post_tags": [
                ""
            ],
            "fields": {
                "title": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "matched_fields": [
                        "title.keyword"
                    ]
                },
                "labels.de-ch.prefix": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true
                    }
                },
                "labels.de.prefix": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true
                    }
                },
                "labels.en.prefix": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true
                    }
                },
                "labels_all.prefix": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true
                    }
                }
            }
        },
        "size": 7,
        "rescore": [
            {
                "window_size": 8192,
                "query": {
                    "query_weight": 0.1,
                    "rescore_query_weight": 1,
                    "score_mode": "total",
                    "rescore_query": {
                        "function_score": {
                            "score_mode": "sum",
                            "functions": [
                                {
                                    "script_score": {
                                        "script": {
                                            "inline": "pow(doc['incoming_links'].value , 2) \/ ( pow(doc['incoming_links'].value, 2) + pow(50,2))",
                                            "lang": "expression"
                                        }
                                    },
                                    "weight": 1.2
                                },
                                {
                                    "script_score": {
                                        "script": {
                                            "inline": "pow(doc['sitelink_count'].value , 2) \/ ( pow(doc['sitelink_count'].value, 2) + pow(20,2))",
                                            "lang": "expression"
                                        }
                                    },
                                    "weight": 1
                                }
                            ]
                        }
                    }
                }
            }
        ],
        "stats": [
            "wikibase-prefix"
        ]
    },
    "options": {
        "timeout": "20s"
    }
}